test-suite-java/src/test/java/io/micronaut/sourcegen/example/InnerTypesTest.java

```diff
@@ -0,0 +1,254 @@
+package io.micronaut.sourcegen.example;
+
+import io.micronaut.inject.processing.ProcessingException;
+import io.micronaut.inject.visitor.VisitorContext;
+import io.micronaut.sourcegen.generator.SourceGenerator;
+import io.micronaut.sourcegen.generator.SourceGenerators;
+import io.micronaut.sourcegen.model.ClassDef;
+import io.micronaut.sourcegen.model.EnumDef;
+import io.micronaut.sourcegen.model.InterfaceDef;
+import io.micronaut.sourcegen.model.ObjectDef;
+import io.micronaut.sourcegen.model.PropertyDef;
+import io.micronaut.sourcegen.model.RecordDef;
+import io.micronaut.sourcegen.model.TypeDef;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class InnerTypesTest {
+
+    private static void assertAndClean(File classBuilder, String expectedString) throws IOException {
```

Comment:
This test probably belongs in `sourcegen-generator/src/main/test/io/micronaut/sourcegen/write`. See other tests there.